apps/src/utils.js

```diff
@@ -431,3 +433,32 @@ exports.ellipsify = function (inputText, maxLength) {
   }
   return inputText || '';
 };
+
+/**
+ * Returns deep merge of two objects, concatenating rather than overwriting
+ * array properites. Does not mutate either object.
+ *
+ * Note: new properties in overrides are always added to end, not in-order.
+ *
+ * Note: may become default behavior of mergeDeep in future immutable versions.
+ *   @see https://github.com/facebook/immutable-js/issues/406
+ *
+ * @param {Object} baseObject
+ * @param {Object} overrides
+ * @returns {Object} original object (now modified in-place)
+ */
+exports.deepMergeConcatArrays = (baseObject, overrides) => {
+  function deepConcatMerger(a, b) {
+    const isList = Immutable.List.isList;
+    if (isList(a) && isList(b)) {
+      return a.concat(b);
+    }
+    if (a && a.mergeWith) {
+      return a.mergeWith(deepConcatMerger, b);
+    }
+    return b;
+  }
+
+  var baseImmutable = Immutable.fromJS(baseObject);
+  return baseImmutable.mergeWith(deepConcatMerger, overrides).toJS();
```

Comment:
To get `_.mergeWith` we'll need lodash > 4.x. With that, getting incompatible version warning for current versions of `grunt-config-watch`, `grunt-config-uglify`, and a dependency of `grunt-cli`.

Upgrading lodash and upgrading those 3 packages leaves 39 unit test fails. Would likely want to use something like https://www.npmjs.com/package/lodash-migrate to scan for potential incompatibilities in the code base, but since not everything is unit test covered it will probably warrant a more careful look at existing usages & UI test run.

Since this is a live site issue and hoping to get this out quickly, & we don't yet have another use case for a full lodash 4.x upgrade, and I'm really doing this just to extend one object and one array, I might lean towards (1) just doing this object merge & extend manually for this case, (2) using immutable as in current PR, or (3) introducing something standalone like [deepmerge](https://github.com/KyleAMathews/deepmerge), though it has issues with mutability of nested properties, is looking for a new owner & isn't super widely used.

Leaning towards #1 or #2, any preference?